// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package meterusage;

// Interface exported by the server.
service MeterUsage {
  // Simple RPC - a request gets a response.
  //
  // Returns all the data within the meterusage timeseries db.
  rpc GetAllRecords (AllRecordsRequest) returns (AllRecordsResponse);
}

// The request message indicating all data required..
message AllRecordsRequest {
  int64 client_id = 1;
  string request_data = 2;
}

// The response message containing the meterusage data
message AllRecordsResponse {
  int64 server_id = 1;
  string response_data = 2;
}

// TODO Generate from within venv and proto folder...
// > python -m grpc_tools.protoc --proto_path=. --python_out=. --grpc_python_out=. meterusage.proto